x-pack/plugins/rollup/public/crud_app/sections/job_create/steps/components/field_chooser.js

```diff
@@ -43,41 +60,77 @@ export class FieldChooser extends Component {
     });
   };
 
-  onSelect = (field) => {
-    this.props.onSelectField(field);
-  }
-
   render() {
     const {
       buttonLabel,
       columns,
       fields,
+      prompt,
+      onSelectField,
     } = this.props;
 
-    const { isOpen } = this.state;
+    const { isOpen, searchValue } = this.state;
 
-    const button = (
-      <EuiButton
-        iconType="arrowDown"
-        iconSide="right"
-        onClick={this.onButtonClick}
-      >
-        {buttonLabel}
-      </EuiButton>
-    );
+    const getRowProps = (item) => {
+      return {
+        onClick: () => {
+          onSelectField(item);
+        },
+      };
+    };
+
+    const searchedItems = searchValue ? fields.filter(item => (
+      item.name.toLowerCase().includes(searchValue.trim().toLowerCase())
+    )) : fields;
+
+    let flyout;
+
+    if (isOpen) {
+      flyout = (
+        <EuiFlyout
+          onClose={this.close}
+          aria-labelledby="fieldChooserFlyoutTitle"
+          size="m"
+          maxWidth={400}
```

Comment:
wonder if it would make sense to add `ownFocus` prop? I keep clicking outside the boundaries of the flyout to attempt to close it, particularly on the Metrics selection where I want to select my aggs after selecting fields

Reply:
Yeah, that's a good point.

Reply:
I decided to leave this as-is for now because I think the UX on a larger screen is pretty nice; as you select items they're added to the table and you can do a visual comparison between the two. We can gauge user feedback and adjust in a later iteration.

![image](https://user-images.githubusercontent.com/1238659/46229734-74729600-c31b-11e8-9076-275f73c799be.png)

Reply:
Another option we can explore later is to use `ownFocus` and to put a checkmark next to each item in the flyout as the user clicks it. Then the user can click a "Add selected terms" button at the bottom of the flyout. Might be a pleasant UX.